core/src/test/java/org/elasticsearch/action/update/UpdateResponseTests.java

```diff
@@ -0,0 +1,132 @@
+            String output = Strings.toString(updateResponse);
+            assertEquals("{\"_index\":\"index\",\"_type\":\"type\",\"_id\":\"id\",\"_version\":0,\"result\":\"not_found\"," +
+                    "\"_shards\":{\"total\":0,\"successful\":0,\"failed\":0}}", output);
+        }
+        {
+            UpdateResponse updateResponse = new UpdateResponse(new ReplicationResponse.ShardInfo(10, 6),
+                    new ShardId("index", "index_uuid", 1), "type", "id", 3, 1, DELETED);
+            String output = Strings.toString(updateResponse);
+            assertEquals("{\"_index\":\"index\",\"_type\":\"type\",\"_id\":\"id\",\"_version\":1,\"result\":\"deleted\"," +
+                    "\"_shards\":{\"total\":10,\"successful\":6,\"failed\":0},\"_seq_no\":3}", output);
+        }
+        {
+            BytesReference source = new BytesArray("{\"title\":\"Book title\",\"isbn\":\"ABC-123\"}");
+            Map<String, GetField> fields = new HashMap<>();
+            fields.put("title", new GetField("title", Collections.singletonList("Book title")));
+            fields.put("isbn", new GetField("isbn", Collections.singletonList("ABC-123")));
+
+            UpdateResponse updateResponse = new UpdateResponse(new ReplicationResponse.ShardInfo(3, 2),
+                    new ShardId("books", "books_uuid", 2), "book", "1", 7, 2, UPDATED);
+            updateResponse.setGetResult(new GetResult("books", "book", "1", 2, true, source, fields));
+
+            String output = Strings.toString(updateResponse);
+            assertEquals("{\"_index\":\"books\",\"_type\":\"book\",\"_id\":\"1\",\"_version\":2,\"result\":\"updated\"," +
+                    "\"_shards\":{\"total\":3,\"successful\":2,\"failed\":0},\"_seq_no\":7,\"get\":{\"found\":true," +
+                    "\"_source\":{\"title\":\"Book title\",\"isbn\":\"ABC-123\"},\"fields\":{\"isbn\":[\"ABC-123\"],\"title\":[\"Book " +
+                    "title\"]}}}", output);
+        }
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        // Create a random UpdateResponse and converts it to XContent in bytes
+        UpdateResponse updateResponse = randomUpdateResponse();
+        BytesReference updateResponseBytes = toXContent(updateResponse, xContentType);
+
+        // Parse the XContent bytes to obtain a parsed UpdateResponse
+        UpdateResponse parsedUpdateResponse;
+        try (XContentParser parser = createParser(xContentType.xContent(), updateResponseBytes)) {
+            parsedUpdateResponse = UpdateResponse.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+
+        // We can't use equals() to compare the original and the parsed update response
+        // because the random update response can contain shard failures with exceptions,
+        // and those exceptions are not parsed back with the same types.
```

Comment:
I changed that part to do something slightly different. We can't compare `UpdateResponse` using `equals()` because it can have shard failures AND the `GetResult` object embedded in the update response is a bit different after parsing (it does not have index/type/id/version).

For the shard failures, we already have `IndexResponseTests.assertDocWriteResponse()` which takes care of checking shard failures, so I reused it.

For the GetResult, I used something similar to the `GetResultTests` where it generates and returns a Tuple of actual and expected `GetResult`, and I set to null the index/type/etc values expected for the parsed GetResult.